src/context/user_context.js

```diff
@@ -0,0 +1,114 @@
+import React, { useContext, useEffect, useState } from 'react';
+import axios, { AxiosError } from 'axios';
+
+// Since this rootURL is used throughout the app, it might make sense to instead have a file for "Constants" that can be imported wherever needed
+const rootUrl = 'https://ecommerce-6kwa.onrender.com';
+
+const UserContext = React.createContext();
+
+export const UserProvider = ({ children }) => {
```

Comment:
This function returns a `Provider` component, with the values that we want to share across the app.